data_steward/tools/create_combined_backup_dataset.py

```diff
@@ -0,0 +1,540 @@
+
+    impersonation_creds = auth.get_impersonation_credentials(
+        args.run_as_email, common.CDR_SCOPES)
+
+    client = BigQueryClient(args.project_id, credentials=impersonation_creds)
+
+    combined_dataset = f'{args.release_tag}_combined_backup'
+    LOGGER.info(
+        f'Creating Combined backup dataset: `{client.project}.{combined_dataset}` '
+    )
+    combined_backup_desc = f'combined raw version of {args.rdr_dataset} + {args.unioned_ehr_dataset}'
+    labels = {
+        "phase": "backup",
+        "release_tag": args.release_tag,
+        "de_identified": "false"
+    }
+    combined_dataset_object = client.define_dataset(combined_dataset,
+                                                    combined_backup_desc,
+                                                    labels)
+    client.create_dataset(combined_dataset_object)
+    LOGGER.info(f'Created dataset `{client.project}.{combined_dataset}`')
+    LOGGER.info('Creating destination CDM tables...')
+    create_cdm_tables(client, combined_dataset)
+
+    LOGGER.info(
+        f'Copying vocabulary tables from {args.vocab_dataset} to {combined_dataset}'
+    )
+    client.copy_dataset(f'{args.vocab_dataset}', f'{combined_dataset}')
+    LOGGER.info(
+        f'Finished Copying vocabulary tables from {args.vocab_dataset} to {combined_dataset}'
+    )
+
+    LOGGER.info('EHR + RDR combine started')
+    LOGGER.info('Verifying all CDM tables in EHR and RDR datasets...')
+    assert_ehr_and_rdr_tables(client, args.rdr_dataset,
+                              args.unioned_ehr_dataset)
+    ehr_consent(client, args.rdr_dataset, combined_dataset)
+
+    for table in combine_consts.RDR_TABLES_TO_COPY:
+        LOGGER.info(f'Copying {table} table from RDR...')
+        client.copy_table(f'{client.project}.{args.rdr_dataset}.{table}',
+                          f'{client.project}.{combined_dataset}.{table}')
+
+    LOGGER.info(f'Translating {combine_consts.PERSON_TABLE} table from EHR...')
+    for table in combine_consts.EHR_TABLES_TO_COPY:
+        LOGGER.info(f'Copying {table} table from EHR...')
+        copy_ehr_table(client, table, args.unioned_ehr_dataset,
+                       combined_dataset)
+
+    for domain_table in combine_consts.DOMAIN_TABLES:
+        LOGGER.info(f'Mapping {domain_table}...')
+        mapping(client, domain_table, args.rdr_dataset,
+                args.unioned_ehr_dataset, combined_dataset)
+    for domain_table in combine_consts.DOMAIN_TABLES:
+        LOGGER.info(f'Loading {domain_table}...')
+        load(client, domain_table, args.rdr_dataset, args.unioned_ehr_dataset,
+             combined_dataset)
+
+    LOGGER.info('Loading fact_relationship...')
+    load_fact_relationship(client, args.rdr_dataset, args.unioned_ehr_dataset,
+                           combined_dataset)
+    LOGGER.info('Loading foreign key Mapped Person table...')
+    load_mapped_person(client, combined_dataset)
```

Comment:
I thought the `person` table in use in the combined data stage came from the rdr dataset.  What am I misunderstanding?

Reply:
This is to update the foreign keys once new primary keys are generated.